### PR TITLE
Update to hca2scea tool: automatically obtain fastq file paths or SRA file paths

### PR DESCRIPTION
**Pull request by ami-day**
This is a pull request for an update to the hca-to-scea-tools repo. In particular, the main "script.py" script, and also the addition of a "fetch_fastq_path.py" module. 
The update enables the script to automatically get the fastq file paths or SRA file paths based on the list of run accessions and also the relevant file names. This information is automatically added into the output sdrf file.

### CODE REVIEW

**Review by amnonkhen — CHANGES_REQUESTED** [review]
Thanks, @ami-day, for the PR.

A few comments:

1. How are the new features tested? I did not notice any change to the test files.
2. Keeping the code as it is, albeit maybe functionally correct, would be incurring a big technical debt. It would be very difficult for us as a team, and even for you who wrote this, to maintain, modify, fix and extend it with confidence. 
3. One of the test scenarios is currently failing.

**Comment by ami-day** [comment]
> Thanks, @ami-day, for the PR.
> 
> A few comments:
> 
> 1. How are the new features tested? I did not notice any change to the test files.
> 2. Keeping the code as it is, albeit maybe functionally correct, would be incurring a big technical debt. It would be very difficult for us as a team, and even for you who wrote this, to maintain, modify, fix and extend it with confidence.
> 3. One of the test scenarios is currently failing.

Thanks for the quick review @amnonkhen, I agree, I'll next work on adding feature tests for this and also spend some time refactoring the code so it is maintainable. I'll work on this this sprint as I'm on development. Will have a look at why 1 of the tests is failing.

File: hca2scea-backend/fetch_fastq_path.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    pool = multiprocessing.Pool(*args, **kwargs)
+    yield pool
+    pool.terminate()
+
+def parse_xml(xml_content):
+    for experiment_package in xml_content.findall('EXPERIMENT_PACKAGE'):
+        yield experiment_package
+
+def filter_paths(sdrf, paths):
+    runs = list(sdrf['Comment[ENA_RUN]'])
```
Comment (amnonkhen) [inline hca2scea-backend/fetch_fastq_path.py:20]: cleanup tip:
This a classic TMI method (Too Much Information)
There is too much detail here.
I would extract bits of it to properly named functions.
For example, the read1 bits, the read2 bits, processing a run.
Would you like to do a few examples together?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: hca2scea-backend/fetch_fastq_path.py (hunk 1 above)
Comment (ami-day) [reply hca2scea-backend/fetch_fastq_path.py:20]: Yes, this is the type of thing I'd love feedback on. I'm going to finish my 1st session of re-organising/refactoring the code, and then going through examples like this to improve further would be great.

Comment (ami-day) [reply hca2scea-backend/fetch_fastq_path.py:20]: I haven't yet refactored the code for this particular new functionality. I have approached the entire script.

**Comment by ami-day** [comment]
Hi @amnonkhen I am finished with my refactoring and added some checks for expected output at the end. It seems to me the tests are working with updated expected output files. I'd like to get this pull request merged soon so I can move onto updating it with new features. Shall we go through how you would approach improving the code first? What's best for you?

p.s. it is saying below that all checks have failed. But, when I run the test script with the characteristic test, it is working fine. Im not sure why this is?

**Pull request merged**